pulp_service/pulp_service/app/urls.py

```diff
@@ -10,13 +11,18 @@
     VulnerabilityReport,
 )
 
+router = SimpleRouter(trailing_slash=False)
+router.register(r"^api/pulp/vuln_report/(?P<uuid>.+)/", VulnerabilityReport, basename="vuln-report")
+
 urlpatterns = [
     path("api/pulp-admin/", admin_site.urls),
     path("api/pulp/redirect-check/", RedirectCheck.as_view()),
     path("api/pulp/internal-server-error-check/", InternalServerErrorCheck.as_view()),
     path("api/pulp/raise-exception-check/", InternalServerErrorCheckWithException.as_view()),
     path("api/pulp/debug_auth_header/", DebugAuthenticationHeadersView.as_view()),
     path("api/pulp/admin/tasks/", TaskViewSet.as_view({"get": "list"})),
-    path("api/pulp/vuln_report/", VulnerabilityReport.as_view({"get": "list", "post": "post"})),
-    path("api/pulp/vuln_report/<uuid:uuid>/", VulnerabilityReport.as_view({"get": "get"})),
+    path(
+        "api/pulp/vuln_report/", VulnerabilityReport.as_view({"get": "retrieve", "post": "create"})
+    ),
+    path("", include(router.urls)),
```

Comment:
I realized something in doing this. We need these urls to be "domainized". These existing urls are for us as admins and we have admin permissions to operate on them, but end-users don't. Also we need the data, e.g. when listing `VulnerabilityReport` objects to be isolated per-domain. As-such, we need to make the model and viewset domain-based...

I'm asking in upstream #pulp-dev on matrix some for examples...